### ISSUE TITLE
Order of pages in the tree view changes from 1.9.1 to 1.9.2

### ISSUE THREAD

**Issue by nilsleiffischer** [doxygen/doxygen#8767]
**Describe the bug**

With Doxygen 1.9.1 the pages in the tree view are listed in the order I add them to `INPUT`. With Doxygen 1.9.2 they are ordered alphabetically.

**Expected behavior**

I need to control the order of the pages, e.g. to have "Installation" before "Contributing guide", so alphabetic ordering doesn't work. Controlling the order with the `INPUT` option in the Doxyfile is also only a workaround, but at least it works (until 1.9.1). A more robust configuration would be even better.

**To Reproduce**

Here is a MWE that demonstrates the problem:

- [doxygen-bug.zip](https://github.com/doxygen/doxygen/files/7121623/doxygen-bug.zip)

**Version**

Doxygen 1.9.1 vs 1.9.2 on macOS 11.2.3 installed via Homebrew, and also on Ubuntu 20.04 compiled from source.

**Comment by doxygen** [comment]
@nilsleiffischer If you want to have files/pages be processed in a certain order you need to specify them explicitly, i.e. use

```
INPUT = docs/PageIndex.dox \
        docs/PageB.dox \
        docs/PageA.dox
```
For directories the order is unspecified. See also https://en.cppreference.com/w/cpp/filesystem/directory_iterator.
In version 1.9.1 and earlier `readdir()` was used, but also there the order is unspecified (but possibly different). So you were relying on unspecified behavior already.

**Comment by nilsleiffischer** [comment]
@doxygen Thanks for your response! Indeed this works when _all_ files are specified explicitly. But as soon as I add a directory to the list, the ordering breaks, even if it's just a directory with source code, which is unavoidable. Here's an extended snippet from the MWE attached above:

```sh
# Doxyfile

# Expectation: Page B should appear before Page A, as defined by
# PageIndex.dox. This works with Doxygen 1.9.1, but doesn't with 1.9.2:
INPUT = docs/PageIndex.dox docs
# Here are some variations that work with Doxygen 1.9.2, but this is very
# fragile: _All_ files must be listed explicitly, and the ordering breaks as
# soon as a directory appears, which is unavoidable when adding source code.
INPUT = docs/PageIndex.dox docs/PageA.md docs/PageB.md        # works
INPUT = docs/PageB.md docs/PageA.md                           # works
INPUT = docs/PageB.md docs/PageA.md docs                      # breaks
INPUT = docs/PageB.md docs/PageA.md src                       # also breaks, but unavoidable!
```

The very last case just adds an unrelated `src` directory, but also breaks the ordering of the pages. Is there a way to order the pages, and also parse source code? The above worked in 1.9.1.

**Comment by doxygen** [comment]
@nilsleiffischer ok, I see what you mean. I'll look into it.

**Comment by nilsleiffischer** [comment]
It does indeed. Thank you for taking care of this so quickly 👍

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
